dashboard/app/models/follower.rb

```diff
@@ -35,9 +35,11 @@ def teacher_must_be_teacher
     errors.add(:user, "must be a teacher") unless user.user_type == User::TYPE_TEACHER
   end
 
-  validate :cannot_follow_yourself, :teacher_must_be_teacher
+  validate :cannot_follow_yourself, unless: -> {deleted?}
+  validate :teacher_must_be_teacher, unless: -> {deleted?}
```

Comment:
Though these are validations we'd like to keep... If the follower is deleted, the teacher may be deleted, in which case `user` is `nil` (scoped to non-deleted accounts), meaning `user.id` and `user.teacher?` would throw.